GitUI/LeftPanel/RepoObjectsTree.cs

```diff
@@ -565,6 +565,17 @@ private void TxtBranchCriterion_KeyDown(object sender, KeyEventArgs e)
             e.Handled = true;
         }
 
+        public override bool ProcessHotkey(Keys keyData)
+        {
+            // skip hotkeys processing when inside branch search control to ensure proper editing
+            if (ActiveControl == _txtBranchCriterion)
```

Comment:
Other hotkeys, particularly the focus-changing `Ctrl+<digit>` should still work.
The following variant should be clear without a comment:
```suggestion
            if (_txtBranchCriterion.ContainsFocus && IsTextEditKey(keyData))
```

Reply:
Thanks, this looks better! Done.